src/http.zig

```diff
@@ -364,6 +369,34 @@ fn NewHTTPContext(comptime ssl: bool) type {
             return @as(*BoringSSL.SSL_CTX, @ptrCast(this.us_socket_context.getNativeHandle(true)));
         }
 
+        pub fn deinit(this: *@This()) void {
+            this.us_socket_context.deinit(ssl);
+            uws.us_socket_context_free(@as(c_int, @intFromBool(ssl)), this.us_socket_context);
```

Comment:
I think there's likely a UAF lurking here. We need to make sure that there are no actively open sockets that belong to this context at the time this is called. The easiest way to do that would be to disable HTTP keep-alive when a custom ca/cert/key is passed.

Reply:
hmm, true. Although disabling keep-alive will probably hurt performance substantially. I thought about `deinit`ing the socket context but that would also close any other open sockets and I assume (I'm guessing here) it would abruptly close any open requests on those sockets, although it would remove the UAF. Not sure what the best option is here while maintaining perf.

I also can't find anywhere `deinit` is actually called, which is probably also an issue

Reply:
I've disabled keepalive in the client when using custom sockets. I don't have a huge amount of spare time to work on this so I'll focus on getting this working atm